ResearchKit/ActiveTasks/ORKDeviceMotionRecorder.m

```diff
@@ -186,23 +188,29 @@ @implementation ORKDeviceMotionRecorderConfiguration
 
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wobjc-designated-initializers"
-- (instancetype)initWithFrequency:(double)freq {
-    self = [super ork_init];
+
+- (instancetype)initWithIdentifier:(NSString *)identifier {
+    @throw [NSException exceptionWithName:NSGenericException reason:@"Use subclass designated initializer" userInfo:nil];
+}
+
+- (instancetype)initWithIdentifier:(NSString *)identifier frequency:(double)freq {
+    self = [super initWithIdentifier:identifier];
     if (self) {
         _frequency = freq;
     }
     return self;
 }
+
 #pragma clang diagnostic pop
 
 - (ORKRecorder *)recorderForStep:(ORKStep *)step outputDirectory:(NSURL *)outputDirectory
 {
-    return [[ORKDeviceMotionRecorder alloc] initWithFrequency:self.frequency
-                                                        step:step
-                                             outputDirectory:(NSURL *)outputDirectory];
+    return [[ORKDeviceMotionRecorder alloc] initWithIdentifier:self.identifier
+                                                     frequency:self.frequency
+                                                          step:step
+                                               outputDirectory:(NSURL *)outputDirectory];
```

Comment:
It doesn't, it's a common mistake I have spotted around. It happens when you copy the method declaration and paste it elsewhere to call it but then forget to delete some argument types. ;)

I'll remove the cast.